From 3a07bae7e8539955bc3a0f4bafe82a8ff0e25fa9 Mon Sep 17 00:00:00 2001
From: Daniel Bristot de Oliveira <daniel@bristot.me>
Date: Thu, 25 Jun 2015 16:51:59 -0300
Subject: [PATCH 2/2] docs: upgrade tuna.8 man page with option -r
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Update tuna.8 man page with the descriptions of the new command line
options:
       -r, --run="COMMAND"
	      Run the COMMAND. If arguments are passed, the entire com‐
	      mand line must be provided inside "quotes". Modifiers  -c
	      and -p can be used to set the affinity and scheduler tun‐
	      ables of the given COMMAND. The arg[0] (i.e. the command)
	      will be set in THREAD-LIST.  Likewise the -t, the COMMAND
	      accepts the prefix + and - as wildcards in  order  to  be
	      appended or removed from THREAD-LIST, respectively.

And the modifications of -p behavior:
       -p, --priority=[POLICY:]RTPRIO
	      Set thread scheduler tunables: POLICY and RTPRIO.  POLICY
	      is one of OTHER, FIFO, RR, or BATCH. If -t is not set, -p
	      will act as a Modifier, saving  the  sched  tunables  for
	      future Actions, for example -r.

And -t behavior:
       -t, --threads=THREAD-LIST
	      THREAD-LIST  affected by commands. Requires a thread num‐
	      ber or thread name, or a comma-separated list  of  thread
	      numbers and/or names. Thread names may contain wildcards.
	      Be sure to quote or escape any  wildcard  specifications.
	      If  only - is passed as argument, the THREAD-LIST will be
	      cleared.

Signed-off-by: Daniel Bristot de Oliveira <daniel@bristot.me>
Signed-off-by: John Kacur <jkacur@redhat.com>
---
 docs/tuna.8 | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/docs/tuna.8 b/docs/tuna.8
index a0a504cb13b9..81a42dd114a5 100644
--- a/docs/tuna.8
+++ b/docs/tuna.8
@@ -35,7 +35,7 @@ Allow all threads to run on CPU\-LIST. Requires \fB\-c\fR or \fB-S\fR.
 Move selected entities to CPU\-LIST. Requires \fB\-c\fR and either \fB-t\fR or \fB-q\fR.
 .TP
 \fB\-p\fR, \fB\-\-priority\fR=\fI[POLICY:]\fRRTPRIO
-Set thread scheduler tunables: POLICY and RTPRIO. POLICY is one of OTHER, FIFO, RR, or BATCH. Requires \fB\-t\fR.
+Set thread scheduler tunables: POLICY and RTPRIO. POLICY is one of OTHER, FIFO, RR, or BATCH. If \fB\-t\fR is not set, \fB\-p\fR will act as a Modifier saving the sched tunables for future Actions, for example \fB\-r\fR.
 .TP
 \fB\-P\fR, \fB\-\-show_threads\fR
 Show thread list.
@@ -43,6 +43,9 @@ Show thread list.
 \fB\-s\fR, \fB\-\-save\fR=\fIFILENAME\fR
 Save kthreads sched tunables to FILENAME.
 .TP
+\fB\-r\fR, \fB\-\-run\fR=\fI"COMMAND"\fR
+Run the COMMAND. If arguments are passed, the entire command line must be provided inside "quotes". Modifiers \fB-c\fR and \fB-p\fR can be used to set the affinity and scheduler tunables of the given COMMAND. The arg[0] (i.e. the command) will be set in THREAD\-LIST. Likewise the \fB-t\fR, the COMMAND accepts the prefix \fB+\fR and \fB-\fR as wildcards in order to be appended or removed from THREAD\-LIST, respectively.
+.TP
 \fB\-v\fR, \fB\-\-version\fR
 Show version
 .TP
@@ -75,7 +78,7 @@ IRQ\-LIST affected by commands. Requires an IRQ number or a comma-separated list
 CPU\-SOCKET\-LIST affected by commands. Requires a socket number or a comma-separated list of socket numbers.
 .TP
 \fB\-t\fR, \fB\-\-threads\fR=\fITHREAD\-LIST\fR
-THREAD\-LIST affected by commands. Requires a thread number or thread name, or a comma-separated list of thread numbers and/or names. Thread names may contain wildcards. Be sure to quote or escape any wildcard specifications.
+THREAD\-LIST affected by commands. Requires a thread number or thread name, or a comma-separated list of thread numbers and/or names. Thread names may contain wildcards. Be sure to quote or escape any wildcard specifications. If only \fB-\fR is passed as argument, the THREAD\-LIST will be cleared.
 .TP
 \fB\-U\fR, \fB\-\-no_uthreads\fR
 Operations will not affect user threads.
-- 
2.4.11

